lib/liquid/standardfilters.rb

```diff
@@ -10,7 +10,7 @@ module StandardFilters
       '"'.freeze => '&quot;'.freeze,
       "'".freeze => '&#39;'.freeze
     }
-    HTML_ESCAPE_ONCE_REGEXP = /["><']|&(?!([a-zA-Z]+|(#\d+));)/
+    HTML_ESCAPE_ONCE_REGEXP = /["><']|&(?!([a-zA-Z]+\d*|(#\d+)|(#[xX][\da-fA-F]+));)/
```

Comment:
Yeah, totally agree. If named entities that contains number are allowed in the HTML spec we should fix both Liquid and Rails.